### PR TITLE
:innocent: extend empty search

### PR DESCRIPTION
**Pull request by beseven**

### CODE REVIEW

File: app/lib/searchValidator.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ function validateSearch(search) {
   let errorMessage = null;
   let searchToReturn = search;
 
-  if (!search) {
+  if ((!search) || ((search) && (search.trim() === ''))) {
```
Comment (neilbmclaughlin) [inline app/lib/searchValidator.js]: Maybe just `if (search && search.trim() !== "" )` to give a less complicated condition and reverse the if/else statements?

Even better make intent explicit by pulling the condition out into a function `if (notEmpty(search)) ..` then you can reuse it below.

File: test/integration/resultsPage.js [hunk 2]
```diff
@@ -40,10 +40,48 @@ describe('Results page', () => {
 describe('Results page error handling', () => {
   describe('when search is not included', () => {
     it('should return a descriptive error messages', (done) => {
+      const search = null;
+      const errorMessage = messages.emptySearchMessage();
```
Comment (neilbmclaughlin) [inline test/integration/resultsPage.js]: It's a matter of taste but I always prefer not using production code in my assertions. I'd prefer to to have the string explicitly stated here (e.g. "You didn't enter any text into the search box. You must type at least part of your GP surgery's name before searching.").

Doing so means I don't have to look anywhere else to see what the message actually is. Also if two errors combine (messages.emptySearchMessage is set to "" and error handling fails then the test won't incorrectly pass. Code duplication in this instance is a good thing.

I know others do not agree with this so feel free to accept, reject or discuss.

Comment (beseven) [inline test/integration/resultsPage.js]: Maybe for tests we should be more explicit than implicit. This does mean the tests will fail when we make content changes, which forces us to change the tests. If we accept that this is how we roll, I'm happy to change it.

File: app/lib/searchValidator.js (hunk 1 above)
Comment (beseven) [inline app/lib/searchValidator.js]: I've gone with `isEmpty(search)` as that is what I'm trying to asses, will push up the code in a bit.

File: test/integration/resultsPage.js (hunk 2 above)
Comment (beseven) [inline test/integration/resultsPage.js]: On a second thought there is a wider discussion here, would it be for integration tests, for unit tests, for both (we are using it for both at the moment)? Can we postpone this request until we have a clear guideline in testing please?

**Comment by beseven** [comment]
Changes pushed

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-15.dev.c2s.nhschoices.net](http://gp-finder-pr-15.dev.c2s.nhschoices.net)

**Pull request merged**